ui/src/Signin.tsx

```diff
@@ -74,13 +75,28 @@ export class Signin extends PureComponent<Props, State> {
   private checkForLogin = async () => {
     try {
       await client.users.me()
+      const redirectIsSet = !!window.localStorage.getItem('redirectTo')
+      if (redirectIsSet) {
+        window.localStorage.removeItem('redirectTo')
+      }
     } catch (error) {
       const {
         location: {pathname},
       } = this.props
 
       clearInterval(this.intervalID)
 
+      if (CLOUD && isFlagEnabled('redirectto')) {
+        const url = new URL(
+          `${window.location.origin}${CLOUD_SIGNIN_PATHNAME}?redirectTo=${
+            window.location.href
+          }`
+        )
+        window.localStorage.setItem('redirectTo', window.location.href)
```

Comment:
why set the `redirectTo` to the current url instead of just removing it from localStorage since we're done with it?

Reply:
Great question - so this feature is actually handling redirectTo logic for three separate environments conditions:

1. Social Sign On is not enabled (tools cluster), and the user logs in via the universal login page
2. Social Sign On **is** enabled and the user is authenticated in Auth0 but not IDPE
3. Social Sign On **is* enabled and the user is **NOT** authenticated in Auth0

By setting the query parameter directly to the initial `api/v2/signin` request, we handle the first two cases (where a user is being navigated - whether directly or silently - to the universal login page). 

The reason we are setting the `redirectTo` onto localStorage is due to the way that authentication occurs in cloud. Basically, by the time we get navigated to the `/login` page, we will have no reference or session with the original redirectTo logic (and the browser will have been redirected about 4 times)

Reply:
>and the browser will have been redirected about 4 times

😬 😱

Reply:
Tell me about it. Things are in need of a good unification